test-integ/catalogv2/simple_test.go

```diff
@@ -0,0 +1,332 @@
+
+// TODO(rb): bump the testing/deployer default versions when the various components have
+// released versions
+func getImages() topology.Images {
+	if *flagUseDevImages {
+		return dev1_17Images
+	}
+	images := utils.TargetImages()
+	images.Dataplane = dev1_17Images.Dataplane
+	return images
+}
+
+func TestSimpleTopologySuite(t *testing.T) {
+	suite.Run(t, new(SimpleTopologySuite))
+}
+
+type SimpleTopologySuite struct {
+	suite.Suite
+	sprawl *sprawl.Sprawl
+}
+
+var _ suite.SetupAllSuite = (*SimpleTopologySuite)(nil)
+
+func (s *SimpleTopologySuite) SetupSuite() {
+	cfg := s.topologyConfig(s.T())
+	s.sprawl = sprawltest.Launch(s.T(), cfg)
+}
+
+func (s *SimpleTopologySuite) TestBasicSingleportExplicitDestination() {
+	t := s.T()
+
+	var (
+		sp       = s.sprawl
+		asserter = topoutil.NewAsserter(sp)
+
+		topo    = sp.Topology()
+		cluster = topo.Clusters["dc1"]
+
+		ships = topoutil.ComputeRelationships(topo)
+	)
+
+	clientV1, err := sp.APIClientForNode("dc1", cluster.FirstServer().ID(), "")
+	require.NoError(t, err)
+
+	clientV2 := sp.ResourceServiceClientForCluster(cluster.Name)
+
+	topoutil.RenderRelationships(ships)
+
+	// Make sure things are truly in v2 not v1.
+	for _, name := range []string{
+		"single-server",
+		"single-client",
+		"multi-server",
+		"multi-client",
+	} {
+		libassert.CatalogServiceDoesNotExist(t, clientV1, name, nil)
+		libassert.CatalogServiceDoesNotExist(t, clientV1, name+"-sidecar-proxy", nil)
+		libassert.CatalogV2ServiceHasEndpointCount(t, clientV2, name, nil, 1)
+	}
+
+	check := func(t *testing.T, ship topoutil.Relationship) {
+		var (
+			svc           = ship.Caller
+			u             = ship.Upstream
+			clusterPrefix string
+		)
+
+		if u.Peer == "" {
+			if u.ID.PartitionOrDefault() == "default" {
+				clusterPrefix = dotjoin(u.PortName, u.ID.Name, u.ID.Namespace, u.Cluster, "internal")
+			} else {
+				clusterPrefix = dotjoin(u.PortName, u.ID.Name, u.ID.Namespace, u.ID.Partition, u.Cluster, "internal-v1")
+			}
+		} else {
+			clusterPrefix = dotjoin(u.ID.Name, u.ID.Namespace, u.Peer, "external")
+		}
+
+		asserter.UpstreamEndpointStatus(t, svc, clusterPrefix+".", "HEALTHY", 1)
+		asserter.HTTPServiceEchoes(t, svc, u.LocalPort, "")
+		asserter.FortioFetch2FortioName(t, svc, u, cluster.Name, u.ID)
+	}
+
+	// Check relationships
+	for _, ship := range ships {
+		t.Run("relationship: "+ship.String(), func(t *testing.T) {
+			check(t, ship)
+		})
+	}
+}
+
+func (s *SimpleTopologySuite) topologyConfig(t *testing.T) *topology.Config {
+	const clusterName = "dc1"
+
+	servers := topoutil.NewTopologyServerSet(clusterName+"-server", 3, []string{clusterName, "wan"}, nil)
+
+	cluster := &topology.Cluster{
+		Enterprise: utils.IsEnterprise(),
+		Name:       clusterName,
+		Nodes:      servers,
+	}
+
+	lastNode := 0
+	nodeName := func() string {
+		lastNode++
+		return fmt.Sprintf("%s-box%d", clusterName, lastNode)
+	}
+
+	s.topologyConfigAddNodes(t, cluster, nodeName, "default", "default")
+	if cluster.Enterprise {
```

Comment:
The same test when executed in enterprise will create additional tenancy microcosms. The behavior of the `relationship` code above will correctly deduce all of these relationships and run the extra tests free of charge.